cirq/ops/controlled_gate.py

```diff
@@ -68,23 +106,28 @@ def _unitary_(self) -> Union[np.ndarray, NotImplementedType]:
         sub_matrix = protocols.unitary(self.sub_gate, None)
         if sub_matrix is None:
             return NotImplemented
-        return linalg.block_diag(np.eye(sub_matrix.shape[0]), sub_matrix)
+        for _ in range(self.total_control_qubits):
```

Comment:
We can very easily compute the size of both blocks, so we probably shouldn't loop.

Reply:
return linalg.block_diag(np.eye(sub_matrix.shape[0] << (self.total_control_qubits - 1)),] sub_matrix)
caused some tests to fail. Will look into this a bit more

Reply:
You should be able to copy over my code from https://github.com/quantumlib/Cirq/pull/1388

Reply:
Yes, that's what I did for the above comment. It still doesn't work. There's a shape mismatch sometimes.